standards/SPIFFE-ID.md

```diff
@@ -74,6 +75,10 @@ Paths MAY be hierarchical - similar to filesystem paths. The specific meaning of
 
   ```spiffe://example.com/9eebccd2-12bf-40a6-b262-65fe0487d453```
 
+### 2.3. Maximum SPIFFE ID Length
+
+URIs, as defined by [RFC 3986](https://tools.ietf.org/html/rfc3986), do not have a maximal length. As an interoperability consideration, SPIFFE implementations MUST support SPIFFE URIs of at least 2048 bytes in length, but MAY support longer. All URI components contribute to the URI length, including the "spiffe" scheme, "://" separator, trust domain, and path component. Note that [RFC 3986](https://tools.ietf.org/html/rfc3986) defines a maximum length of 255 characters for the "host" component of a URI; therefore a maximum length of a trust domain is 255 bytes.
```

Comment:
If we are going to say that implementations must support at least 2048 bytes, then we are also setting the practical maximum length (rather than the minimum) because once an ID is greater than 2048 bytes all interop bets are off.

I think that the section title helps the reader understand this point, but it would also help to be more explicit about this. I suggest amending the text with something like the following:
`
... at least 2048 bytes in length, but MAY support longer. Note that, as a result, SPIFFE IDs greater than 2048 bytes in length may encounter interoperability problems. Operators are strongly encouraged to limit SPIFFE IDs to a maximum of 2048 bytes in length.
`

Of course, this begs the question of whether should we be setting a maximum size here instead of a minimum size, and mandate that implementations support the full length? It seems like flipping it around might make this section easier to read and understand. I am not sure if there are use cases for larger sizes, but it does feel like once 2048 bytes are crossed, SPIFFE is immediately less valuable.

Reply:
@evan2645 Could you say anything more about "once 2048 bytes are crossed, SPIFFE is immediately less valuable"?

Reply:
> once 2048 bytes are crossed, SPIFFE is immediately less valuable

What I meant here is that the main goal of SPIFFE is to provide interoperable identity - if we state that all bets are off once crossing the 2048 limit, then going above that limit seems contrary to our goal.

For posterity: this topic was discussed in the last SIG-Spec call. The participants (which included myself and @justinburke) reached consensus on defining a maximum length that all implementations must support, rather than a minimum supported length. Taking this approach will eliminate the above-2048 grey area, and simplify the language in this document.